middle_end/flambda2/bound_identifiers/bound_for_function.ml

```diff
@@ -42,12 +42,17 @@ let[@ocamlformat "disable"] print ppf
 
 let create ~return_continuation ~exn_continuation ~params ~my_closure ~my_depth
     =
-  let params_set = BP.Set.of_list params in
-  if List.length params <> BP.Set.cardinal params_set
+  BP.List.check_no_duplicates params;
```

Comment:
Shouldn't the no duplicate check be inside the `check_invariants ()` block ?
EDIT: so the `check_no_duplicates` function already polls the `check_invariant` function, but it's a bit confusing...

Reply:
Yeah, but I think it's probably ok.